common/utils/src/main/resources/error/error-classes.json

```diff
@@ -75,6 +75,16 @@
       }
     }
   },
+  "AVRO_INCORRECT_TYPE" : {
```

Comment:
shall we unify the error class? `AVRO_INCOMPATIBLE_READ_TYPE`

Reply:
We can just say that the type is incompatible and users can set the legacy config to restore the legacy undefined behavior.

Reply:
sure!